sass/_4.header.scss

```diff
@@ -0,0 +1,104 @@
+.header {
+	background: radial-gradient(rgb(194, 74, 126), rgb(105, 86, 235));
+	display: flex;
+	justify-content: center;
+	padding: 1rem 1rem 15rem 1rem;
+	position: relative;
+
+	.nav {
+		align-items: center;
+		display: flex;
+		height: calc(3vw + 3vh);
+		justify-content: space-between;
+
+		.nav__logo {
+			width: 7rem;
+		}
+
+		&__links {
```

Comment:
Jak wyżej pisałem, zrobiłbym Block `.header` + klasy do niego należące i osobno `nav` - będzie czytelniej.
W BEM nie ograniczamy styli względem dodatkowego blocku. Jeśli potrzebujemy specjalnych cech to wykorzystujemy `--modyfikator`. Tutaj znajdziesz więcej informacji: https://devmentor.pl/b/metodologia-bem-w-css-i-sass